sim/pokemon.js

```diff
@@ -89,6 +89,10 @@ class Pokemon {
 		/**@type {string | boolean} */
 		this.moveThisTurn = '';
 
+		// For Stomping Tantrum
+		this.moveLastTurnSucceeded = null;
+		this.moveThisTurnSucceeded = null;
```

Comment:
Does this really need both? In theory one `lastMoveSucceeded` flag should be enough, right? Replace at the end of move execution (end of `runMove` is probably a good place)?

Reply:
According to http://www.smogon.com/forums/threads/ultra-sun-ultra-moon-battle-mechanics-research-read-post-2.3620030/page-5#post-7589613 it actually checks the last move used on the previous turn and not any move used this turn.

Reply:
Doing it in `runMove` wouldn’t work anyway because moves used through things like Magic Bounce count

Reply:
I would have a slight preference for the term `Result` as opposed to `Succeeded` as the latter sounds too boolean when in fact `null` is not a failure.

Reply:
Me too, but one more question: Is there any solid evidence that the game is tracking two moves simultaneously? Like, Stomping Tantrum conclusively reacting to a move last turn, when a move earlier this turn did something else?

Reply:
The post I linked pretty explicitly describes that exact scenario, so I think what you meant to say was @Marty-D

Reply:
It describes it badly, and doesn't really differentiate between the possibilities:

1. At least one success since the last turn's move is enough
2. Instructed moves are ignored

Reply:
Also: Is `null` really not a failure? That's not good.

Reply:
Same guy confirmed earlier that instructed moves do count. 
`undefined` works too

Reply:
http://www.smogon.com/forums/threads/ultra-sun-ultra-moon-battle-mechanics-research-read-post-2.3620030/page-4#post-7581284

Reply:
That still doesn't answer my question. All of this testing but no one's really testing the logic behind it.

Reply:
"Instruct, Dancer, Magic Coat, Magic Bounce double ST BP on failure" doesn't help! It's incredibly ambiguous!

turn 1: move failed
turn 2: Magic Bounce move succeeded
turn 2: ST double or not?

turn 1: move succeeded
turn 2: Magic Bounce move failed
turn 2: ST double or not?